src/components/structures/UserSettings.js

```diff
@@ -631,6 +632,17 @@ module.exports = React.createClass({
         }
     },
 
+    onInterfaceScaleChange: function(newInterfaceScale) {
+        newInterfaceScale = parseInt(newInterfaceScale);
+        if (this.state.interfaceScale !== newInterfaceScale) {
+            SettingsStore.setValue("interfaceScale", null, SettingLevel.DEVICE, newInterfaceScale);
+            this.setState({
+                interfaceScale: newInterfaceScale,
+            });
+            PlatformPeg.get().reload();
```

Comment:
Changing scale yielding a full reload is really bad UX :(

Reply:
Yes. Just a copy from language setting to start with. What would you prefer happen? Options:
1. Note that zoom will apply after a restart of the app
2. Zoom as the setting is changed - might be buggy because of scaling happening while the user is touching controls
3. Zoom with a delay?

If 2 is preferable, is there a way in this sdk to watch a state change (scale changing) on the root element? Maybe `SettingsManager` has a watch function? Or is the only way to bubble up the change through the component tree?

Reply:
you could use the dispatcher to send a `zoom_change` event that LoggedInView/MatrixChat listens on and acts on